.github/workflows/main-version.yaml

```
@@ -0,0 +1,109 @@
+          # Fetch tags from the main branch
+          git fetch origin main --tags
+          # Get the latest tag from the main branch
+          TAG=$(git tag --list --sort=-v:refname | grep -E '^v[0-9]+\.[0-9]+\.[0-9]+' | head -n 1 || echo "0.0.0")
+          # Remove the 'v' prefix from the tag
+          CLEAN_TAG=$(echo $TAG | sed 's/^v//')
+          echo "LATEST_TAG=$CLEAN_TAG" >> $GITHUB_ENV
+          echo "::set-output name=latest_tag::$CLEAN_TAG"
+
+      - name: Compare versions
+        id: compare_versions
+        run: |
```

actionlint (1.7.7)

workflow command "set-output" was deprecated. use `echo "{name}={value}" >> $GITHUB_OUTPUT` instead: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions

(deprecated-commands)